pandas/tseries/tests/test_timeseries.py

```diff
@@ -2454,6 +2455,83 @@ def test_intersection_bug_1708(self):
         result = index_1 & index_2
         self.assertEqual(len(result), 0)
 
+    # GH 10699
+    def test_datetime64_with_DateOffset(self):
+        for klass in (Series, DatetimeIndex):
+            s = klass(date_range('2000-01-01', '2000-01-31'))
+            result = s + pd.DateOffset(years=1)
+            result2 = pd.DateOffset(years=1) + s
+            exp = klass(date_range('2001-01-01', '2001-01-31'))
+            self.assert_numpy_array_equal(result, exp)
+            self.assert_numpy_array_equal(result2, exp)
+
+            result = s - pd.DateOffset(years=1)
+            exp = klass(date_range('1999-01-01', '1999-01-31'))
+            self.assert_numpy_array_equal(result, exp)
+
+            s = klass([Timestamp('2000-01-15 00:15:00', tz='US/Central'),
+                       pd.Timestamp('2000-02-15', tz='US/Central')])
+            result = s + pd.offsets.MonthEnd()
+            result2 = pd.offsets.MonthEnd() + s
+            exp = klass([Timestamp('2000-01-31 00:15:00', tz='US/Central'),
+                         Timestamp('2000-02-29', tz='US/Central')])
+            self.assert_numpy_array_equal(result, exp)
+            self.assert_numpy_array_equal(result2, exp)
+
+            # array of offsets - valid for Series only
+            if klass is Series:
+                s = klass([Timestamp('2000-1-1'), Timestamp('2000-2-1')])
+                result = s + Series([pd.offsets.DateOffset(years=1),
+                                    pd.offsets.MonthEnd()])
+                exp = klass([Timestamp('2001-1-1'), Timestamp('2000-2-29')])
+                self.assert_numpy_array_equal(result, exp)
+
```

Comment:
add an array of the *same* offset (just to show it works, in theory it could be done in a vectorized way, but as apply ok for now). also these will raise warnings, use ``assert_raises_warning(PerformanceWarning)`` where needed